modules/xml/test/src/smithy4s/xml/XmlCodecSpec.scala

```diff
@@ -481,43 +492,73 @@ object XmlDecodingSpec extends SimpleIOSuite {
     val xml = """|<Bar>
                  |   <x>1</x>
                  |</Bar>""".stripMargin
-    decodeDocument[Foo](xml).attempt.map { result =>
-      expect.same(
-        result,
-        Left(
-          XmlDecodeError(XPath.root, "Expected Foo XML root element, got Bar")
+    parseDocument(xml)
+      .flatMap(decodeDocument[Foo](_))
+      .attempt
+      .map { result =>
+        expect.same(
+          result,
+          Left(
+            XmlDecodeError(XPath.root, "Expected Foo XML root element, got Bar")
+          )
         )
-      )
-    }
+      }
   }
 
-  def checkContent[A: Schema](xml: String, expected: A)(implicit
+  def checkContent[A: Schema](xmlString: String, expected: A)(implicit
```

Comment:
Switched the logic to test both encoding and decoding at the same time. The tests were already containing nearly everything we needed for that purpose.

Reply:
can we label them somehow to make errors clearer? i.e.g label one as an assertion on encoding, the other on decoding

Reply:
Weaver doesn't have a combinator for this and I don't want to make a bespoke one. However we have `traced(here)` which will capture the accurate position of the expectation (source locations are kept in a list in weaver, so you can trace the calls reasonably accurately)

Reply:
[d45899d](https://github.com/disneystreaming/smithy4s/pull/503/commits/d45899dc33844c1b601d57e2076470c2af4b1a94)